oshmem/shmem/c/shmem_mcs_lock.c

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2013      Mellanox Technologies, Inc.
+ *                         All rights reserved.
+ * Copyright (c) 2014      Cisco Systems, Inc.  All rights reserved.
+ * Copyright (c) 2015      Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * $COPYRIGHT$
+ *
```

Comment:
as this is newly introduced file, i'd guess only relevant Nvidia copyright notice should be present

Reply:
Fixed the License